src/main.rs

```diff
@@ -225,10 +235,27 @@ fn settings() -> io::Result<Settings> {
         None => DEFAULT_DELETION_CHUNK_SIZE,
     };
 
+    // validates the input argument parsed into --min-age and converts input to a duration
+    let min_age = match matches.value_of(MIN_AGE_OPTION) {
+        Some(value) => {
+            let regex = Regex::new(r"(\d+)([smhd])").unwrap();
+            if regex.is_match(value) {
+                match parse(value) {
+                    Ok(duration) => Some(duration),
+                    Err(_) => None
```

Comment:
We shouldn't silently ignore parse errors.

Reply:
Error handling implemented